Vector_graphics_on_surfaces/include/CGAL/Vector_graphics_on_surfaces/locally_shortest_path.h

```diff
@@ -0,0 +1,4814 @@
+ * \param directions contains the direction one need to move from `center` to reach each vertex of the polygon.
+ * \param lengths the distance one need to move from `center` along the direction at the same position in `directions` to reach each vertex of the polygon.
+ * \param tmesh input triangle mesh supporting the vertices of the output polygon
+ * \param solver container for the precomputed information. If not initialized, it will be initialized internally.
+ * \return a face location for each vertex of the polygon
+ * \todo add named parameters
+ * \todo polygon orientation is not handled in the function and should be done outside of the function for now
+ * \todo offer something better than a 2D vector for the direction
+ * \todo directly handle polyline?
+ * \todo why the first polygon vertex is duplicated by the function? (most probably for the example but it shouldn't be done here)
+ */
+template <class K, class TriangleMesh>
+std::vector<CGAL::Polygon_mesh_processing::Face_location<TriangleMesh,typename K::FT>>
+trace_geodesic_polygon(const CGAL::Polygon_mesh_processing::Face_location<TriangleMesh, typename K::FT> &center,
+                       const std::vector<typename K::Vector_2>& directions,
+                       const std::vector<typename K::FT>& lengths,
+                       const TriangleMesh &tmesh,
+                       const Dual_geodesic_solver<typename K::FT>& solver = {})
+{
+  namespace PMP = CGAL::Polygon_mesh_processing;
+
+  std::size_t n=directions.size();
+  std::vector<PMP::Face_location<TriangleMesh,typename K::FT>> result;
+  std::vector<PMP::Face_location<TriangleMesh, typename K::FT>> vertices(n);
+#ifdef CGAL_DEBUG_BSURF
+  std::cout << "trace_geodesic_polygon\n";
+#endif
+  for(std::size_t i=0;i<n;++i)
+  {
+    vertices[i]= straightest_geodesic<K>(center,directions[i],lengths[i],tmesh).back();
+#ifdef CGAL_DEBUG_BSURF
+    std::cout << PMP::construct_point(vertices[i], tmesh) << "\n";
+#endif
+  }
+  std::vector<PMP::Edge_location<TriangleMesh,typename K::FT>> edge_locations;
+
+  const Dual_geodesic_solver<typename K::FT>* solver_ptr=&solver;
+  Dual_geodesic_solver<typename K::FT> local_solver;
+  if (solver.graph.empty())
+  {
+    solver_ptr = &local_solver;
+    init_geodesic_dual_solver(local_solver, tmesh);
+  }
+
+  for(std::size_t i=0;i<n;++i)
+  {
+    edge_locations.clear();
+    locally_shortest_path<typename K::FT>(vertices[i],vertices[(i+1)%n],tmesh, edge_locations, *solver_ptr);
+    result.push_back(vertices[i]);
+    for(auto& el : edge_locations)
+    {
+      result.push_back(PMP::to_face_location(el, tmesh));
+    }
+  }
+  result.push_back(vertices[0]);
+
+  return result;
+}
+
+/*!
+ * \ingroup VGSFunctions
+ * computes for each vertex of each polygon in `polygons` a face location on `tmesh`, `center` representing the center of the 2D bounding box of the polygons.
```

Comment:
Is this the axis aligned or the oriented bounding box?   I guess it is the latter, but as `Bbox_2`  does not have axis aligned in the name, lets be more explicit here.